academia/admin.py

```diff
@@ -1,47 +1,131 @@
-    list_filter = ('country', 'type', 'owned_by', 'unlisted')
-    empty_value_display = '-empty field-'
+    list_display = (
+        "name",
+        "country",
+        "code",
+        "type",
+        "owned_by",
+        "unlisted",
+    )
+    search_fields = (
+        "name__icontains",
+        "country__name__icontains",
+        "code__icontains",
+        "type__icontains",
+        "owned_by__icontains",
+        "country__country_code__icontains",
+        "country__continent__icontains",
+    )
+    list_filter = ("country", "type", "owned_by", "unlisted")
+    empty_value_display = "-empty field-"
+
 
 class AdminFaculty(admin.ModelAdmin):
-    list_display = ('name', 'school',)
-    search_fields = ['name__icontains', 'school__name__icontains']
-    list_filter = ('school',)
-    empty_value_display = '-empty field-'
-    autocomplete_fields = ['school']
-    
+    list_display = (
+        "name",
+        "school",
+    )
+    search_fields = ["name__icontains", "school__name__icontains"]
+    list_filter = ("school",)
+    empty_value_display = "-empty field-"
+    autocomplete_fields = ["school"]
+
+
 class AdminDepartment(admin.ModelAdmin):
-    list_display = ('name', 'duration', 'school', 'faculty')
-    search_fields = ('name__icontains', 'degree__name__icontains', 'degree__code__icontains', 'duration__icontains', 'faculty__name__icontains', 'school__name__icontains', 'school__country__name__icontains', 'school__country__country_code__icontains')
-    list_filter = ('degree', 'faculty', 'duration')
-    empty_value_display = '-empty field-'
-    autocomplete_fields = ['school', 'faculty', 'degree']
+    list_display = ("name", "duration", "school", "faculty")
+    search_fields = (
+        "name__icontains",
+        "degree__name__icontains",
+        "degree__code__icontains",
+        "duration__icontains",
+        "faculty__name__icontains",
+        "school__name__icontains",
+        "school__country__name__icontains",
+        "school__country__country_code__icontains",
+    )
+    list_filter = ("degree", "faculty", "duration")
+    empty_value_display = "-empty field-"
+    autocomplete_fields = ["school", "faculty", "degree"]
+
 
 class AdminDegree(admin.ModelAdmin):
-    list_display = ('name', 'code')
-    search_fields = ('name__icontains', 'code__icontains')
-    list_filter = ('name', 'code')
-    empty_value_display = '-empty field-'
+    list_display = ("name", "code")
+    search_fields = ("name__icontains", "code__icontains")
+    list_filter = ("name", "code")
+    empty_value_display = "-empty field-"
+
+
+class AdminClient(admin.ModelAdmin):
+    list_display: list = ["id", "name", "email", "is_verified", "client_type"]
+    list_filter: list = ["is_verified", "client_type"]
+
+
+class AdminClientAPIKey(admin.ModelAdmin):
```

Comment:
Although I see it structured here, I couldn't create a ClientAPIKey in the Django Admin.